src/libostree/ostree-repo-commit.c

```diff
@@ -614,30 +614,32 @@ write_content_object (OstreeRepo         *self,
                                     cancellable, error))
     return FALSE;
 
-  gboolean temp_file_is_regular = g_file_info_get_file_type (file_info) == G_FILE_TYPE_REGULAR;
-  gboolean temp_file_is_symlink = g_file_info_get_file_type (file_info) == G_FILE_TYPE_SYMBOLIC_LINK;
-  gboolean object_is_symlink = temp_file_is_symlink;
+  gboolean phys_object_is_symlink = FALSE;
+  const GFileType object_file_type = g_file_info_get_file_type (file_info);
+  switch (object_file_type)
+    {
+    case G_FILE_TYPE_REGULAR:
+      break;
+    case G_FILE_TYPE_SYMBOLIC_LINK:
+      if (self->mode == OSTREE_REPO_MODE_BARE || self->mode == OSTREE_REPO_MODE_BARE_USER_ONLY)
+        phys_object_is_symlink = TRUE;
+      break;
+    default:
+      return glnx_throw (error, "Unsupported file type %u", object_file_type);
+    }
 
-  if (repo_mode == OSTREE_REPO_MODE_BARE_USER && object_is_symlink)
+  /* For bare-user, convert the symlink target to the input stream */
+  if (repo_mode == OSTREE_REPO_MODE_BARE_USER && object_file_type == G_FILE_TYPE_SYMBOLIC_LINK)
     {
       const char *target_str = g_file_info_get_symlink_target (file_info);
       g_autoptr(GBytes) target = g_bytes_new (target_str, strlen (target_str) + 1);
 
-      /* For bare-user we can't store symlinks as symlinks, as symlinks don't
-         support user xattrs to store the ownership. So, instead store them
-         as regular files */
-      temp_file_is_regular = TRUE;
-      temp_file_is_symlink = FALSE;
-
       if (file_input != NULL)
         g_object_unref (file_input);
       /* Include the terminating zero so we can e.g. mmap this file */
```

Comment:
We do write it actually, because we drop into the `g_output_stream_splice()` path.  But our `fallocate()` is wrong.

I'm actually kind of dubious about the value of writing it, since it doesn't make sense to mmap small files.  But let's keep it.  Fixup and new commit :arrow_down: